backend/src/main/kotlin/me/andrewda/utils/Models.kt

```diff
@@ -0,0 +1,109 @@
+package me.andrewda.utils
+
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.jetbrains.exposed.dao.Entity
+import org.jetbrains.exposed.dao.EntityID
+import kotlin.reflect.full.memberProperties
+
+enum class ReadLevel {
+    USER, SELF, ADMIN
+}
+
+/**
+ * Marks a field as readable in the API response.
+ *
+ * @param key the key to use in the response (defaults to the variable name)
+ * @param readLevel the [ReadLevel] required to read this value
+ * @param deep whether this field is a deep (performs a database query)
+ */
+annotation class Readable(
```

Comment:
It would be better to avoid reflection entirely. All we need to do is make the key field mandatory. This also makes the user of this annotation conscious of how the field of the readable object interacts with the request. Finally, it creates a layer of separation between the client request and the name of the property field, which is always good. That means that if someone renames the variable, our entire API doesn't suddenly break.

Reply:
Maybe... But reflection would still be necessary to get the values of the properties and to get the `id` field which isn't annotated.